### PR TITLE
add: return-book functionality with validate

### PR DESCRIPTION
**Pull request by Nyxan**
without return
Знімок екрана 2024-06-16 193601
return-book page
Знімок екрана 2024-06-16 193626
after used post
Знімок екрана 2024-06-16 193651
after changes
Знімок екрана 2024-06-16 193742
error validate
Знімок екрана 2024-06-16 193724

### CODE REVIEW

File: borrowing_app/views.py [hunk 1]
```diff
@@ -36,7 +39,31 @@ def get_queryset(self):
...
         return self.serializer_class
 
     def perform_create(self, serializer):
         serializer.save(user=self.request.user)
+
+    @action(
+        detail=True,
+        methods=["get", "post"],
+        permission_classes=[IsAuthenticated],
+        url_path="return-book",
```
Comment (nicksetrakov) [inline borrowing_app/views.py]: image

**Review by nicksetrakov — CHANGES_REQUESTED** [review]

File: borrowing_app/views.py (hunk 1 above)
Comment (nicksetrakov) [reply borrowing_app/views.py]: and we don't need GET method for this action if I got it right

**Review by panicua — APPROVED** [review]
seems nice, but before merging, try to return not your borrowing, so you have 2 users, try to return other users borrwing with your endpoint and see what happens.

**Review by SashaKisliy — APPROVED** [review]
GJ!

**Review by nicksetrakov — APPROVED** [review]

**Comment by Nyxan** [comment]
user cant return another borrowing (admin can)
if he in url has not his book-id, if he try, changes will not happen
Знімок екрана 2024-06-16 233100

**Pull request merged**